test/drawbridge/client_test.clj

```diff
@@ -0,0 +1,39 @@
+(ns drawbridge.client-test
+  (:require [clojure.test :refer [deftest is use-fixtures testing]]
+            [compojure.core :refer [ANY defroutes]]
+            [compojure.handler :as handler]
+            [drawbridge.core :as drawbridge]
+            [drawbridge.client]
+            [nrepl.core :as nrepl]
+            [org.httpkit.server :as server]))
+
+(let [nrepl-handler (drawbridge/ring-handler)]
+  (defroutes app
+    (ANY "/repl" request (nrepl-handler request))))
+
+(defn server-fixture
+  [f]
+  (let [stop-fn (server/run-server (handler/api #'app) {:port 12345})]
+    (f)
+    (stop-fn)))
+
+(use-fixtures :once server-fixture)
+
+(defn send-message
+  [client message]
+  (-> client
+      (nrepl/message message)
+      nrepl/response-values
+      last))
+
+(deftest sending-messages
+  (with-open [conn (nrepl/url-connect "http://localhost:12345/repl")]
+    (let [client (nrepl/client conn 1000)]
+      (testing "Evaluating valid form returns the result"
+        (is (= 3 (send-message client {:op "eval" :code "(+ 1 2)"}))))
```

Comment:
I think it'd be better to compare the entire response map, especially if the cases of errors.

Reply:
Actually, I just noticed you're using `response-values`, so that's not a big deal.

Reply:
Yeah, that's true. It's a bit difficult to figure out what would be the best level to test on.

Anyway, I refactored the tests to compare response maps instead. I agree that the error cases especially are clearer now. 

The response maps have `id` and `session` keys which have UUID values, so I excluded those from the test. Also the entire response seq has map entries like this:

```clojure
{:err "RuntimeException EOF while reading, starting at line 1  clojure.lang.Util.runtimeException (Util.java:221)\n"}
```

That would tie the tests to the line numbers in Clojure core, so I figured it might be best to leave them out of the comparison, too.

Anyway, let me know if you'd like me to make any more changes or whether you preferred the previous style of testing.